nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/PutMarkLogic.java

```diff
@@ -226,53 +235,76 @@ public void init(ProcessorInitializationContext context) {
     @OnScheduled
     public void onScheduled(ProcessContext context) {
         super.populatePropertiesByPrefix(context);
+
         dataMovementManager = getDatabaseClient(context).newDataMovementManager();
+
         writeBatcher = dataMovementManager.newWriteBatcher()
             .withJobId(context.getProperty(JOB_ID).getValue())
             .withJobName(context.getProperty(JOB_NAME).getValue())
             .withBatchSize(context.getProperty(BATCH_SIZE).asInteger())
-            .withTemporalCollection(context.getProperty(TEMPORAL_COLLECTION).getValue());
+            .withTemporalCollection(context.getProperty(TEMPORAL_COLLECTION).getValue())
+            .onBatchSuccess(this::onBatchSuccess)
+            .onBatchFailure(this::onBatchFailure);
 
         ServerTransform serverTransform = buildServerTransform(context);
         if (serverTransform != null) {
             writeBatcher.withTransform(serverTransform);
         }
+
         Integer threadCount = context.getProperty(THREAD_COUNT).asInteger();
-        if(threadCount != null) {
+        if (threadCount != null) {
             writeBatcher.withThreadCount(threadCount);
         }
-        this.writeBatcher.onBatchSuccess(writeBatch -> {
```

Comment:
Moved the success/failure code into separate protected methods so it's easier to understand and optionally override.

Reply:
Much easier to understand, thanks.

Reply:
@ehennum I'm planning on adding at least unit tests this week for all of these PRs. Will see about integration tests.

Reply:
Thanks, @rjrudin . If the pull request can provide an appropriate level of coverage (no need for tests for test's sake), that's greatly appreciated.